eva-accession-pipeline/src/test/java/uk/ac/ebi/eva/accession/pipeline/runner/RestartFailedJobTest.java

```diff
@@ -0,0 +1,281 @@
+        // Mock the behavior of shutDownAccessionGenerator method to do nothing
+        doNothing().when(accessioningServiceSpy).shutDownAccessionGenerator();
+
+    }
+
+    @After
+    public void tearDown() {
+        jobRepositoryTestUtils.removeJobExecutions();
+        inputParameters.setForceRestart(false);
+    }
+
+    @Test
+    @DirtiesContext
+    public void restartFailedJobThatIsAlreadyInTheRepository() throws Exception {
+        useTempVcfFile();
+        injectErrorIntoTempVcf();
+        JobInstance failingJobInstance = runJobAandCheckResults();
+
+        mongoTemplate.dropCollection(SubmittedVariantEntity.class);
+
+        inputParameters.setForceRestart(true);
+        remediateTempVcfError();
+        deleteTemporaryContigAndVariantFiles(); //left behind by unsuccessful job A
+        runJobBAndCheckRestart(failingJobInstance);
+    }
+
+    private JobInstance runJobAandCheckResults() throws Exception {
+        runner.run();
+        assertEquals(EvaAccessionJobLauncherCommandLineRunner.EXIT_WITH_ERRORS, runner.getExitCode());
+        JobInstance currentJobInstance = CommandLineRunnerUtils.getLastJobExecution(CREATE_SUBSNP_ACCESSION_JOB,
+                        jobExplorer,
+                        inputParameters.toJobParameters())
+                .getJobInstance();
+        StepExecution stepExecution = jobRepository.getLastStepExecution(currentJobInstance,
+                CREATE_SUBSNP_ACCESSION_STEP);
+        //Ensure that only the first batch was written (batch size is 5 and error was at line#9)
+        assertEquals(inputParameters.getChunkSize(), stepExecution.getWriteCount());
+
+        return currentJobInstance;
+    }
+
+    private void runJobBAndCheckRestart(JobInstance failingJobInstance) throws Exception {
+        runner.run();
+        assertEquals(EvaAccessionJobLauncherCommandLineRunner.EXIT_WITHOUT_ERRORS, runner.getExitCode());
+        JobInstance currentJobInstance = CommandLineRunnerUtils.getLastJobExecution(CREATE_SUBSNP_ACCESSION_JOB,
+                        jobExplorer,
+                        inputParameters.toJobParameters())
+                .getJobInstance();
+        assertNotEquals(failingJobInstance.getInstanceId(), currentJobInstance.getInstanceId());
+    }
+
+    private void injectErrorIntoTempVcf() throws Exception {
+        String modifiedVcfContent = originalVcfContent.replace("76852", "76852jibberish");
+        // Inject error in the VCF file to cause processing to stop at variant#9
+        writeToTempVCFFile(modifiedVcfContent);
+    }
+
+    private void remediateTempVcfError() throws Exception {
+        writeToTempVCFFile(originalVcfContent);
+    }
+
+    private void useOriginalVcfFile() throws Exception {
+        inputParameters.setVcf(originalVcfInputFilePath);
+        vcfReader.setResource(FileUtils.getResource(new File(originalVcfInputFilePath)));
+    }
+
+    private void useTempVcfFile() throws Exception {
```

Comment:
We should be reusing these methods from `EvaAccessionJobLauncherCommandLineRunnerTest`, and perhaps add some documentation in the code as to why we had to separate these particular tests from the main class.